File: 백준/Gold/2749. 피보나치 수 3/피보나치 수 3.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
long long fibo(long long n) {
    int mod = 1000000;
    int p = mod / 10 * 15;
    int *fib = (int *)malloc(sizeof(int) * p);
    memset(fib, 0, sizeof(int) * p);
    fib[1] = 1;
    for (int i = 2; i < p; i++) {
        fib[i] = fib[i - 1] + fib[i - 2];
        fib[i] %= mod;
    }
    return fib[n % p];
}
int main(void) {
    long long n;
    scanf("%lld", &n);
    printf("%lld", fibo(n));
    return 0;
}

File: 백준/Gold/2749. 피보나치 수 3/피보나치 수 3.py
import sys
def fibo(n):
    mod=1000000
    p=mod//10*15
    fib=[0]*p
    fib[1]=1
    for i in range(2,p):
        fib[i]=fib[i-1]+fib[i-2]
        fib[i]%=mod
    return fib[n%p]
n=int(sys.stdin.readline().rstrip())
print(fibo(n))